### PR TITLE
fix missing message index mutation 

### PR DESCRIPTION
**Pull request by bingyuyap**
This PR does a few things:
- fixes missing observation bug where monitor cannot find unprocessed messages as message indexes weren't created in `FlushCache`. 
- logs the bulk mutation for more observability.
- updates `MAX_CHAIN_ID` to be the largest mainnet chain id.

### CODE REVIEW

**Review by lrogana — APPROVED** [review]

File: fly/cmd/historical_uptime/main.go [hunk 1]
```diff
@@ -187,10 +189,7 @@ func initPromScraper(promRemoteURL string, logger *zap.Logger, errC chan error)
 				case <-t.C:
 					recordGuardianHeightDifferences()
 
-					for i := 1; i < 36; i++ {
-						if i == PYTHNET_CHAIN_ID {
-							continue
-						}
+					for i := 1; i <= int(MAX_CHAIN_ID); i++ {
```
Comment (panoel) [inline fly/cmd/historical_uptime/main.go:192]: You are tracking pythnet, then?

**Review by panoel — APPROVED** [review]

File: fly/cmd/historical_uptime/main.go (hunk 1 above)
Comment (bingyuyap) [reply fly/cmd/historical_uptime/main.go:192]: yes, as it seems that the batch processing has enabled us to track pyth without performance degradation

File: fly/cmd/historical_uptime/main.go [hunk 2]
```diff
@@ -109,7 +109,9 @@ var (
 	)
 )
 
-const PYTHNET_CHAIN_ID = int(vaa.ChainIDPythNet)
+// [0, MAX_CHAIN_ID] is the range of chain id that we will track for the uptime monitor
+// in this case it's snaxchain since it's the largest mainnet chain idj
+const MAX_CHAIN_ID = vaa.ChainIDSnaxchain
```
Comment (bruce-riley) [inline fly/cmd/historical_uptime/main.go:114]: This is a maintenance headache. We should think about adding something in the go sdk to give you a max value that will be updated whenever we add a new chain. Or is this meant to match the dashboard / fly? Should we have a shared constant somewhere?

**Review by bruce-riley — APPROVED** [review]

File: fly/cmd/historical_uptime/main.go (hunk 2 above)
Comment (bingyuyap) [reply fly/cmd/historical_uptime/main.go:114]: Will address this in a separate PR

**Pull request merged**